src/nixdimensions.cc

```diff
@@ -1,8 +1,11 @@
 #include <nix.hpp>
-#include "mex.h"
 #include "arguments.h"
 #include "struct.h"
 
+#define char16_t LIBRARY_char16_t
+#include "mex.h"
+#undef char16_t
```

Comment:
Why exactly is this needed? Can you remember? Does `char16_t` get redefined in any later includes (not in this file, but in some other file)? `LIBRARY_char16_t` is never used anywhere (and we are technically not a library) so I guess we don't need this. 
We should probably add a comment explaining why we have to do this. On that matter I would also avoid doing it 4 times. Rather move it to a (possibly new) central include file and do it only once.